mm2src/mm2_main/src/lp_swap/saved_swap.rs

```diff
@@ -140,6 +141,26 @@ impl SavedSwap {
             },
         };
     }
+
+    pub async fn fetch_and_set_usd_prices(&mut self) {
+        if let Some(rates) = fetch_swap_coins_price(
+            Some(self.maker_coin_ticker().unwrap_or_else(|_| "".to_string())).clone(),
+            Some(self.taker_coin_ticker().unwrap_or_else(|_| "".to_string())).clone(),
+        )
+        .await
+        {
+            match self {
+                SavedSwap::Maker(maker) => {
+                    maker.maker_coin_usd_price = Some(rates.base);
+                    maker.taker_coin_usd_price = Some(rates.rel);
+                },
+                SavedSwap::Taker(taker) => {
+                    taker.maker_coin_usd_price = Some(rates.base);
+                    taker.taker_coin_usd_price = Some(rates.rel);
+                },
+            }
```

Comment:
It can be simplifier as
```rust
match self {
  SavedSwap::Maker(maker) => maker.fetch_and_set_usd_prices(),
  SavedSwap::Taker(taker) => taker.fetch_and_set_usd_prices(),
}
```

Reply:
done